### PR TITLE
Add AV checking by examining file tags (added by ClamAV lambda)

### PR DESCRIPTION
**Pull request by mattandrews**
This relies on the lambda function I've set up which scans all our form S3 buckets (dev, test and prod) and tags suspected files as infected. If these tags are found, the file is blocked from being downloaded by users, and not uploaded to Salesforce.

Update: new error message.
Screenshot 2019-07-22 11 03 44

### CODE REVIEW

File: controllers/apply/form-router-next/lib/file-uploads.js [hunk 1]
```diff
@@ -109,6 +109,31 @@ function uploadFile({ formId, applicationId, fileMetadata }) {
...
+function getObjectTagging(key) {
+    return s3
+        .getObjectTagging({
+            Bucket: S3_UPLOAD_BUCKET,
+            Key: key
+        })
+        .promise();
+}
+
+function checkAntiVirus({ formId, applicationId, filename }) {
```
Comment (davidrapson) [inline controllers/apply/form-router-next/lib/file-uploads.js]: You shouldn't need to construct a `new Promise` here or use `async` as you are already returning a promise. Could try something like:

```suggestion
function checkAntiVirus({ formId, applicationId, filename }) {
    const keyName = [formId, applicationId, filename].join('/');
    return s3
        .getObjectTagging({
            Bucket: S3_UPLOAD_BUCKET,
            Key: keyName
        })
        .promise()
        .then(function(tags) {
            const tagSet = get(tags, 'TagSet', []);
            const avStatus = tagSet.find(t => t.Key === 'av-status');
            if (avStatus && get(avStatus, 'Value') === 'INFECTED') {
                logger.error(`Infected file found at ${keyName}`);
                throw new Error('Infected file suspected');
            } else {
                return avStatus;
            }
        });
}
```

File: controllers/apply/form-router-next/index.js [hunk 2]
```diff
@@ -458,39 +473,51 @@ function initFormRouter({
...
+
+                try {
+                    await checkAntiVirus(pathConfig);
+                } catch (err) {
+                    return res.send(
+                        'This file has been blocked for security reasons.'
+                    );
+                }
+
+                getObject(pathConfig)
```
Comment (davidrapson) [inline controllers/apply/form-router-next/index.js]: I'd possibly move this block inside the `try`

File: controllers/apply/form-router-next/lib/file-uploads.js (hunk 1 above)
Comment (mattandrews) [reply controllers/apply/form-router-next/lib/file-uploads.js]: Good call, done

File: controllers/apply/form-router-next/index.js (hunk 2 above)
Comment (mattandrews) [reply controllers/apply/form-router-next/index.js]: 👍

**Comment by mattandrews** [comment]
Now updated so we show an error/warning after submission:

Screenshot 2019-07-19 12 02 29

Screenshot 2019-07-19 12 03 44

**Pull request merged**